### ISSUE TITLE
ERROR  error is not defined

### ISSUE THREAD

**Issue by ndarilek** [facebook/react-native#5309]
Guessing this is an error in my JS but it would be helpful if it reported exactly what the error was. :) 0.18rc.

```

> Perceptron@0.0.1 start /home/nolan/Projects/Perceptron/Perceptron
> node node_modules/react-native/local-cli/cli.js start

 ┌────────────────────────────────────────────────────────────────────────────┐ 
 │  Running packager on port 8081.                                            │ 
 │                                                                            │ 
 │  Keep this packager running while developing on any JS projects. Feel      │ 
 │  free to close this tab and run your own packager instance if you          │ 
 │  prefer.                                                                   │ 
 │                                                                            │ 
 │  https://github.com/facebook/react-native                                  │ 
 │                                                                            │ 
 └────────────────────────────────────────────────────────────────────────────┘ 
Looking for JS files in
   /home/nolan/Projects/Perceptron/Perceptron 

[8:30:15 AM] <START> Building Dependency Graph
[8:30:15 AM] <START> Crawling File System
[8:30:15 AM] <START> Loading bundles layout
[8:30:15 AM] <END>   Loading bundles layout (0ms)

React packager ready.

 ERROR  error is not defined
ReferenceError: error is not defined
    at ChildProcess.<anonymous> (/home/nolan/Projects/Perceptron/Perceptron/node_modules/fb-watchman/index.js:204:15)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:821:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)

See http://facebook.github.io/react-native/docs/troubleshooting.html
for common problems and solutions.

npm ERR! Linux 4.2.8-300.fc23.x86_64
npm ERR! argv "/usr/bin/node" "/home/nolan/.node_modules/bin/npm" "start"
npm ERR! node v5.4.0
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! Perceptron@0.0.1 start: `node node_modules/react-native/local-cli/cli.js start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Perceptron@0.0.1 start script 'node node_modules/react-native/local-cli/cli.js start'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the Perceptron package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node node_modules/react-native/local-cli/cli.js start
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs Perceptron
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls Perceptron
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/nolan/Projects/Perceptron/Perceptron/npm-debug.log
```

Here is npm-debug.log:

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/home/nolan/.node_modules/bin/npm', 'start' ]
2 info using npm@3.5.2
3 info using node@v5.4.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle Perceptron@0.0.1~prestart: Perceptron@0.0.1
6 silly lifecycle Perceptron@0.0.1~prestart: no script for prestart, continuing
7 info lifecycle Perceptron@0.0.1~start: Perceptron@0.0.1
8 verbose lifecycle Perceptron@0.0.1~start: unsafe-perm in lifecycle true
9 verbose lifecycle Perceptron@0.0.1~start: PATH: /home/nolan/.node_modules/lib/node_modules/npm/bin/node-gyp-bin:/home/nolan/Projects/Perceptron/Perceptron/node_modules/.bin:/home/nolan/.go/bin:/home/nolan/.cargo/bin:/home/nolan/opt/android/sdk/tools:/home/nolan/opt/android/sdk/platform-tools:/home/nolan/.node_modules/bin:/home/nolan/.local/bin:/home/nolan/bin:/home/nolan/.go/bin:/home/nolan/.cargo/bin:/home/nolan/opt/android/sdk/tools:/home/nolan/opt/android/sdk/platform-tools:/home/nolan/.node_modules/bin:/home/nolan/.local/bin:/home/nolan/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
10 verbose lifecycle Perceptron@0.0.1~start: CWD: /home/nolan/Projects/Perceptron/Perceptron
11 silly lifecycle Perceptron@0.0.1~start: Args: [ '-c',
11 silly lifecycle   'node node_modules/react-native/local-cli/cli.js start' ]
12 silly lifecycle Perceptron@0.0.1~start: Returned: code: 1  signal: null
13 info lifecycle Perceptron@0.0.1~start: Failed to exec start script
14 verbose stack Error: Perceptron@0.0.1 start: `node node_modules/react-native/local-cli/cli.js start`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/home/nolan/.node_modules/lib/node_modules/npm/lib/utils/lifecycle.js:232:16)
14 verbose stack     at emitTwo (events.js:87:13)
14 verbose stack     at EventEmitter.emit (events.js:172:7)
14 verbose stack     at ChildProcess.<anonymous> (/home/nolan/.node_modules/lib/node_modules/npm/lib/utils/spawn.js:24:14)
14 verbose stack     at emitTwo (events.js:87:13)
14 verbose stack     at ChildProcess.emit (events.js:172:7)
14 verbose stack     at maybeClose (internal/child_process.js:821:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
15 verbose pkgid Perceptron@0.0.1
16 verbose cwd /home/nolan/Projects/Perceptron/Perceptron
17 error Linux 4.2.8-300.fc23.x86_64
18 error argv "/usr/bin/node" "/home/nolan/.node_modules/bin/npm" "start"
19 error node v5.4.0
20 error npm  v3.5.2
21 error code ELIFECYCLE
22 error Perceptron@0.0.1 start: `node node_modules/react-native/local-cli/cli.js start`
22 error Exit status 1
23 error Failed at the Perceptron@0.0.1 start script 'node node_modules/react-native/local-cli/cli.js start'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the Perceptron package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     node node_modules/react-native/local-cli/cli.js start
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs Perceptron
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls Perceptron
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by ndarilek** [comment]
This appears to be a Watchman failure case in that killing it resolves 
the issue. Still, though, would be good if the tool told me as much.

Thanks.

**Comment by mkonicek** [comment]
This is a known issue and will be fixed in 0.18.

**Comment by anil826** [comment]
MAC OS Sierra same problem not fixed yet
